sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpClient.cs

```diff
@@ -120,7 +120,7 @@ internal class AmqpClient : TransportClient
             ServiceBusRetryPolicy retryPolicy,
             string identifier)
         {
-            Argument.AssertNotClosed(_closed, nameof(AmqpClient));
+            Argument.AssertNotDisposed(_closed, nameof(AmqpClient));
```

Comment:
Note: `_closed` here is only set to true in the DisposeAsync of the AmqpClient. So, it is the same as checking if it was disposed.

Reply:
nit: can we just rename to _disposed?

Reply:
In this case, the naming used for the AmqpClient matches that of the TransportClient and the ServiceBusConnection. I also thought about changing the name but it is exposed as `IsClosed`. The other places we call AssertNotDisposed have a `IsDisposed` property instead. I think it is fine if this instance uses it as `_closed`.

Reply:
Those are all internal so we could just rename everything to IsDisposed, but your call.

Reply:
I think that they contextually makes sense for them to relate to closing. Currently, there is a 1-to-1 between disposing and closing (meaning DisposeAsync calls CloseAsync). But, I doubt a lot of this will change. If we go through the effort of pealing layers, it would make sense to remove the closing stuff. But, I don't think it is a big deal either way.